pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +89,51 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano())) //creates a more unique random
+	var err error
+	var taskInfo *types.TaskInfo
+	backoffFactor := int64(1)
+
+	//setting up timer
+	timer := time.NewTimer(time.Duration(100) * time.Millisecond)
+	if !timer.Stop() {
+		<-timer.C
+	}
+
+	for {
+		taskInfo, err = WaitForResult(ctx, f)
+		if err != nil {
+			return nil, err
+		}
+
+		if taskInfo.Error == nil {
+			break
+		}
+
+		if _, ok := taskInfo.Error.Fault.(types.TaskInProgressFault); !ok {
+			log.Debugf("Task failed during a retry operation : %#v", taskInfo.Task)
+			log.Debugf("Failed TaskInfo Object : %#v", taskInfo)
+			return taskInfo, errors.New(taskInfo.Error.LocalizedMessage)
+		}
```

Comment:
can we have a newline after this `if` block to separate an exit path from the retry logic